apps/src/studio/studio.js

```diff
@@ -1653,10 +1679,28 @@ var TICKS_BEFORE_FACE_SOUTH = 5;
  * Given direction/emotion/tickCount, calculate which frame number we should
  * display for sprite.
  */
-function spriteFrameNumber (index) {
+function spriteFrameNumber (index, opts) {
   var sprite = Studio.sprite[index];
   var frameNum = 0;
-  if (sprite.frameCounts.turns === 7 && sprite.displayDir !== Direction.SOUTH) {
+
+  var currentTime = new Date();
+  var elapsed = currentTime - Studio.startTime;
+
+  if (opts && opts.walkDirection) {
+    return frameDirTableWalking[sprite.displayDir];
+  }
+  else if (opts && opts.walkFrame) {
+    if (sprite.timePerFrame) {
```

Comment:
can we just add this conditional to the line above and avoid extra levels on indentation?